pkgs/by-name/ot/otb/package.nix

```
@@ -0,0 +1,125 @@
+{
+  cmake,
+  callPackage,
+  fetchFromGitHub,
+  makeWrapper,
+  lib,
+  stdenv,
+  swig,
+  which,
+  boost,
+  curl,
+  gdal,
+  libsvm,
+  libgeotiff,
+  muparser,
+  muparserx,
+  opencv,
+  perl,
+  python3,
+  shark,
+  tinyxml,
+  enableFeatureExtraction ? true,
+  enableHyperspectral ? true,
+  enableLearning ? true,
+  enableMiscellaneous ? true,
+  enableOpenMP ? false,
+  enablePython ? true,
+  extraPythonPackages ? ps: with ps; [ ],
+  enableRemote ? true,
+  enableSAR ? true,
+  enableSegmentation ? true,
+  enableStereoProcessing ? true,
+}:
+assert enablePython -> python3 != null;
+let
+  inherit (lib) optionalString optionals optional;
+  pythonInputs =
+    optionals enablePython (with python3.pkgs; [ numpy ]) ++ (extraPythonPackages python3.pkgs);
+
+  otb-itk = callPackage ./itk_4_13/package.nix { };
+  otb-shark = shark.override { enableOpenMP = enableOpenMP; };
+
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "otb";
+  version = "9.0.0";
+
+  src = fetchFromGitHub {
+    owner = "orfeotoolbox";
+    repo = "otb";
+    rev = finalAttrs.version;
+    hash = "sha256-Ut2aimQL6Reg62iceoaM7/nRuEV8PBWtOK7KFHKp0ws=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    swig
+    which
+  ];
+
+  buildInputs = [
+    boost
+    curl
+    gdal
+    libsvm
+    libgeotiff
+    muparser
+    muparserx
+    opencv
+    otb-itk
+    otb-shark
+    perl
+    tinyxml
+  ] ++ optionals enablePython [ python3 ] ++ optionals enablePython pythonInputs;
+
+  # https://www.orfeo-toolbox.org/CookBook/CompilingOTBFromSource.html#native-build-with-system-dependencies
+  # activates all modules and python by default
+  cmakeFlags =
+    [ ]
+    ++ optionals enableFeatureExtraction [ "-DOTB_BUILD_FeaturesExtraction=ON" ]
+    ++ optionals enableHyperspectral [ "-DOTB_BUILD_Hyperspectral=ON" ]
```

```suggestion
    ++ optional enableHyperspectral "-DOTB_BUILD_Hyperspectral=ON"
```
we can shorten that a bit

---

I have applied these changes!